pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/AuthenticationRoute.java

```diff
@@ -24,78 +35,158 @@
+        claims.put(userId, auth0Claims.get(userId).asString());
+        //todo pegah get user id from database once DDP-7172 is done
 
-                String jwtToken =
-                        new SecurityHelper().createToken(jwtSecret, cookieAgeInSeconds + (System.currentTimeMillis() / 1000) + (60 * 5),
-                                claims);
+        return claims;
+    }
 
-                DSMToken authResponse = new DSMToken(jwtToken);
-                return authResponse;
-            } else {
-                throw new RuntimeException("UserIdentity not found");
+    private String getNewAuth0TokenWithCustomClaims(Map<String, String> claims, String clientSecret, String clientId, String auth0Domain,
+                                                    String auth0Audience, String audienceNameSpace) throws AuthenticationException {
+        String requestUrl = "https://" + auth0Domain + api;
+        Map<String, String> headers = new HashMap<>();
+        headers.put("content-type", contentType);
+
+        List<NameValuePair> requestParams =
+                buildRequestParams(clientId, clientCredentials, clientSecret, auth0Audience, claims, audienceNameSpace);
+        Auth0M2MResponse response;
+        try {
+            response = DDPRequestUtil.postRequestWithResponse(Auth0M2MResponse.class, requestUrl, requestParams, "auth0 M2M", headers);
+            if (response == null) {
+                throw new AuthenticationException("Didn't receive a token from auth0!");
+            }
+        } catch (Exception e) {
+            throw new AuthenticationException("couldn't get response from Auth0 for user " + claims.get("USER_EMAIL"), e);
+        }
+        if (response.getError() != null) {
+            throw new AuthenticationException("Got Auth0 M2M error " + response.getError() + " : " + response.getErrorDescription());
+        }
+        return response.getAccessToken();
+    }
+
+    private List<NameValuePair> buildRequestParams(@NonNull String clientId, @NonNull String grantType, @NonNull String clientSecret,
+                                                   @NonNull String audience, Map<String, String> claims, String audienceNameSpace) {
+        List<NameValuePair> params = new ArrayList<>();
+        for (String key : claims.keySet()) {
+            String finalKey = key;
+            if (key.indexOf(audienceNameSpace) == -1) {
+                finalKey = audienceNameSpace + key;
             }
-        } else {
-            throw new RuntimeException("There was no token in the payload");
+            params.add(new BasicNameValuePair(finalKey, claims.get(key)));
         }
+        params.add(new BasicNameValuePair(clientIdKey, clientId));
+        params.add(new BasicNameValuePair(grantTypeKey, grantType));
+        params.add(new BasicNameValuePair(clientSecretKey, clientSecret));
+        params.add(new BasicNameValuePair(audienceKey, audience));
+        return params;
     }
 
     private static class DSMToken {
```

Comment:
to make it easier, do we still need the DSMToken class? could we just return the string from method `getNewAuth0TokenWithCustomClaims`